zio-crypto/shared/src/main/scala/zio/crypto/signature/Signature.scala

```diff
@@ -0,0 +1,93 @@
+package zio.crypto.signature
+
+import zio._
+import zio.crypto.random.SecureRandom
+import zio.crypto.random.SecureRandom.SecureRandom
+
+import java.security.{ KeyPairGenerator, PrivateKey, PublicKey, Signature => JSignature }
+
+case class SignatureObject(value: Array[Byte]) extends AnyVal
+sealed class SignatureAlgorithm(val name: String)
+
+object SignatureAlgorithm {
+  case object ECDSASHA256 extends SignatureAlgorithm("SHA256withECDSA")
+  case object ECDSASHA384 extends SignatureAlgorithm("SHA384withECDSA")
+  case object ECDSASHA512 extends SignatureAlgorithm("SHA512withECDSA")
+}
+
+case class SignaturePrivateKey(key: PrivateKey, algorithm: SignatureAlgorithm)
+case class SignaturePublicKey(key: PublicKey, algorithm: SignatureAlgorithm)
+case class SignatureKeyPair(publicKey: SignaturePublicKey, privateKey: SignaturePrivateKey)
+
+object Signature {
+  type Signature = Has[Signature.Service]
+
+  trait Service {
+    def genKey(alg: SignatureAlgorithm): Task[SignatureKeyPair]
+    def sign(m: Array[Byte], privateKey: SignaturePrivateKey): RIO[SecureRandom, SignatureObject]
+    def verify(m: Array[Byte], signature: SignatureObject, publicKey: SignaturePublicKey): Task[Boolean]
+  }
+
+  val live: ULayer[Signature] = ZLayer.succeed(new Service {
+
+    def genKey(alg: SignatureAlgorithm): Task[SignatureKeyPair] = Task.effect {
+      val keyPairGenerator = KeyPairGenerator.getInstance("EC")
```

Comment:
I think setting the key size with the `initialize` method is forgotten here based on ECDSASHA256 or ECDSASHA384 or ECDSASHA512. Am I right?

Reply:
I don't believe so. The algorithm is specified on line 46 in this file.

Reply:
Yes, that's right.